source/components/StyledArticleTagsSelector/index.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+import TagShape from './TagShape';
+import AddedTags from './AddedTags';
+import SuggestedTags from './SuggestedTags';
+import SearchForm from './SearchForm';
+
+const Wrapper = styled.div`
+    font-size: ${({ theme }) => theme.font_size.s};
+`;
+
+/**
+ * `StyledArticleTagsSelector` component.
+ *
+ * **NOTE**: Several props are arrays of `Tag`s - `Tag` is defined as follows:
+ * ```typescript
+ * interface Tag {
+ *   id: string;
+ *   title: string;
+ *   url?: string;
+ * }
+ * ```
+ */
+function StyledArticleTagsSelector({
+    className,
+    communityName,
+    maxAllowed,
+    onAddTag,
+    onRemoveTag,
+    onSearch,
+    onSuggestedTag,
+    searchResults,
+    suggestedTags,
+    tags,
+}) {
+    const [query, setQuery] = React.useState('');
+    const maxNumOfTagsAdded = tags && (tags.length >= maxAllowed);
+
+    const searchCallback = (q) => {
+        setQuery(q);
+        onSearch(q);
+    };
+
+    return (
+        <Wrapper className={className}>
+            <SearchForm
+                communityName={communityName}
+                maxNumOfTagsAdded={maxNumOfTagsAdded}
+                onAddTag={onAddTag}
+                query={query}
+                onSearch={searchCallback}
+                searchResults={searchResults}
+            />
+            <AddedTags tags={tags} onRemove={onRemoveTag} />
+            {!maxNumOfTagsAdded && <SuggestedTags tags={suggestedTags} onClick={onSuggestedTag} />}
```

Comment:
Does it need a special function or can it also use `onAddTag`?

Reply:
Depends - in original fepo code those are two distinct functions so there might be some logic there. Remember that this component is only the visual part - there's no logic in it.